spring-cloud-stream-binder-kafka-streams/src/test/java/org/springframework/cloud/stream/binder/kafka/streams/integration/KafkaStreamsBinderWordCountIntegrationTests.java

```diff
@@ -133,6 +138,18 @@ private void receiveAndValidate(ConfigurableApplicationContext context) throws E
 		assertThat(cr.value().contains("\"word\":\"foobar\",\"count\":1")).isTrue();
 	}
 
+	private void sendTombStoneRecordsAndVerifyGracefulHandling() {
+		Map<String, Object> senderProps = KafkaTestUtils.producerProps(embeddedKafka);
+		DefaultKafkaProducerFactory<Integer, String> pf = new DefaultKafkaProducerFactory<>(senderProps);
+		KafkaTemplate<Integer, String> template = new KafkaTemplate<>(pf, true);
+		template.setDefaultTopic("words");
+		template.sendDefault(null);
+		ConsumerRecords<String, String> received = consumer.poll(Duration.ofMillis(5000));
+		//By asserting that the received record is empty, we are ensuring that the tombstone record
+		//was handled by the binder gracefully.
+		assertThat(received.isEmpty()).isTrue();
```

Comment:
You have to destroy `DefaultKafkaProducerFactory` in the end of class do not pollute a memory with non-used active objects.